lib/asm/arm64/Arm64Assembler.v3

```diff
@@ -0,0 +1,98 @@
+// Copyright 2024 Virgil Authors. All rights reserved.
+// See LICENSE for details of Apache 2.0 license.
+
+// Arm64 registers come in 32-bit and 64-bit variants
+type RegSize {
+	case Half;
+	case Full;
+}
+
+// Base class for all Arm64 registers
+// regnum is an ID 0-32
+class Arm64Reg(regnum: byte) {
+	def low3 = byte.view(regnum & 0b111);
+}
+
+// Representation of an Arm64 general purpose register (x0-x30),
+// or the zero register (xzr), or the stack pointer (sp).
+// An "int reg", is not a standard name in the Arm64 reference, but we use it
+// to distinguish from the SIMD and floating-point register file
+class Arm64IntReg extends Arm64Reg {
+	def name32: string;
+	def name64: string;
+
+	new(name32, name64, regnum: byte) super(regnum) {}
+
+	def name(size: RegSize) -> string {
+		match (size) {
+			Half => return name32;
+			Full => return name64;
+		}
+	}
+}
+
+class Arm64Sfr extends Arm64Reg(0) {} // TODO SIMD + floating point registers
```

Comment:
What does `Sfr` stand for? Can you add this in the comment?

Reply:
sign and floating point register. Fixed it